pkg/operation/botanist/component/istio/istiod_test.go

```diff
@@ -2495,6 +2495,40 @@ spec:
 				Expect(string(managedResourceSecret.Data["istio-ingress_templates_service_test-ingress.yaml"])).To(Equal(istioIngressService(&externalTrafficPolicy)))
 			})
 		})
+
+		Context("VPN disabled", func() {
+			BeforeEach(func() {
+				for i := range igw {
+					igw[i].VPNEnabled = false
+				}
+
+				istiod = NewIstio(
+					c,
+					renderer,
+					Values{
+						Istiod: IstiodValues{
+							Image:       "foo/bar",
+							Namespace:   deployNS,
+							TrustDomain: "foo.local",
+							Zones:       []string{"a", "b", "c"},
+						},
+						IngressGateway:       igw,
+						ProxyProtocol:        ipp,
+						DNSServerAddress:     pointer.String("1.2.3.4"),
+						NodeLocalIPVSAddress: pointer.String("1.2.3.4"),
+					},
+				)
+			})
+
+			It("should successfully deploy all resources", func() {
+				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResourceSecret), managedResourceSecret)).To(Succeed())
+				Expect(managedResourceSecret.Type).To(Equal(corev1.SecretTypeOpaque))
+				Expect(managedResourceSecret.Data).To(HaveLen(42))
```

Comment:
Should the amount of resources not be less than with vpn enabled? I would have expected it to be `40`, i.e. the two resources below should not be present.
WDYT?

Reply:
The file itself still exists in the chart but it's empty if VPN is disabled. This is reflected in the `managedSecretSecret` by the registry and therefore I used this [expectation](https://github.com/gardener/gardener/blob/0f36db392f884d7a54b5fced6dba096e130da375/pkg/operation/botanist/component/istio/istiod_test.go#L2528) for example.